rust/.gitignore

```diff
@@ -0,0 +1,2 @@
+/target
+/dist
```

Comment:
1) The `/dist` here didn't work for me; I wound up with `osm2lanes-web/dist/` showing up as untracked
2) I like specifying the Rust-specific ignores here; would you mind removing `target` from `../.gitignore` and moving `Cargo.lock` here please?

Reply:
- [ ] gitignore fixes and imprrovements